modules/publication-service-oaipmh/src/main/java/org/opencastproject/publication/oaipmh/OaiPmhPublicationServiceImpl.java

```diff
@@ -79,453 +80,648 @@
-
-  /** The security service */
-  private SecurityService securityService = null;
-
-  /** The user directory service */
-  private UserDirectoryService userDirectoryService = null;
-
-  /** The organization directory service */
-  private OrganizationDirectoryService organizationDirectoryService = null;
-
-  /** The download distribution service */
-  private DownloadDistributionService downloadDistributionService = null;
-
-  /** The streaming distribution service */
-  private StreamingDistributionService streamingDistributionService = null;
-
-  /** The OAI-PMH persistence */
-  private OaiPmhDatabase persistence = null;
-
+  private DownloadDistributionService downloadDistributionService;
   private OaiPmhServerInfo oaiPmhServerInfo;
+  private OaiPmhDatabase oaiPmhDatabase;
+  private OrganizationDirectoryService organizationDirectoryService;
+  private SecurityService securityService;
+  private ServiceRegistry serviceRegistry;
+  private StreamingDistributionService streamingDistributionService;
+  private UserDirectoryService userDirectoryService;
 
-  /**
-   * Callback for the OSGi environment to set the service registry reference.
-   *
-   * @param serviceRegistry
-   *          the service registry
-   */
-  protected void setServiceRegistry(ServiceRegistry serviceRegistry) {
-    this.serviceRegistry = serviceRegistry;
-  }
-
-  /**
-   * Callback for setting the security service.
-   *
-   * @param securityService
-   *          the securityService to set
-   */
-  public void setSecurityService(SecurityService securityService) {
-    this.securityService = securityService;
-  }
-
-  /**
-   * Callback for setting the user directory service.
-   *
-   * @param userDirectoryService
-   *          the userDirectoryService to set
-   */
-  public void setUserDirectoryService(UserDirectoryService userDirectoryService) {
-    this.userDirectoryService = userDirectoryService;
+  public OaiPmhPublicationServiceImpl() {
+    super(JOB_TYPE);
   }
 
-  /**
-   * Sets a reference to the organization directory service.
-   *
-   * @param organizationDirectory
-   *          the organization directory
-   */
-  public void setOrganizationDirectoryService(OrganizationDirectoryService organizationDirectory) {
-    this.organizationDirectoryService = organizationDirectory;
+  @Override
+  protected String process(Job job) throws Exception {
+    if (!StringUtils.equalsIgnoreCase(JOB_TYPE, job.getJobType()))
+      throw new IllegalArgumentException("Can not handle job type " + job.getJobType());
+
+    MediaPackage mediaPackage = MediaPackageParser.getFromXml(job.getArguments().get(0));
+    Publication publication = null;
+    boolean checkAvailability = false;
+    switch (Operation.valueOf(job.getOperation())) {
+      case Publish:
+        publication = publish(job, mediaPackage, job.getArguments().get(1),
+                Collections.set(StringUtils.split(job.getArguments().get(2), SEPARATOR)),
```

Comment:
Could you please introduce meaningful constants to access the arguments? That would be more readable.